sycl/gdb/libsycl.so-gdb.py

```diff
@@ -51,35 +51,38 @@ def value(self, arg):
 class HostAccessor(Accessor):
     """For Host device memory layout"""
 
-    def payload(self):
-        return self.obj["impl"]["_M_ptr"].dereference()
-
     def memory_range(self, dim):
-        return self.payload()["MMemoryRange"]["common_array"][dim]
+        eval_string = "((" + str(self.obj.type) + ")" + str(self.obj) + ")->getMemoryRange()"
+        return gdb.parse_and_eval(eval_string)["common_array"][dim];
 
     def offset(self, dim):
-        return self.payload()["MOffset"]["common_array"][dim]
+        eval_string = "((" + str(self.obj.type) + ")" + str(self.obj) + ")->getOffset()"
+        return gdb.parse_and_eval(eval_string)["common_array"][dim];
 
     def data(self):
-        return self.payload()["MData"]
-
+        eval_string = "((" + str(self.obj.type) + ")" + str(self.obj) + ")->getPtr()"
+        return gdb.parse_and_eval(eval_string);
 
 class HostAccessorLocal(HostAccessor):
     """For Host device memory layout"""
 
+    def memory_range(self, dim):
+        eval_string = "((" + str(self.obj.type) + ")" + str(self.obj) + ")->getSize()"
+        return gdb.parse_and_eval(eval_string)["common_array"][dim];
+
     def index(self, arg):
         if arg.type.code == gdb.TYPE_CODE_INT:
             return int(arg)
         result = 0
         for dim in range(self.depth):
             result = (
-                result * self.payload()["MSize"]["common_array"][dim]
-                + arg["common_array"][dim]
+                result * self.memory_range() + arg["common_array"][dim]
             )
         return result
 
     def data(self):
-        return self.payload()["MMem"]
+        eval_string = "((" + str(self.obj.type) + ")" + str(self.obj) + ")->getPtr()"
+        return gdb.parse_and_eval(eval_string);
```

Comment:
This method now looks the same as the parent's implementation. Can it be removed?

Reply:
Yes, removed.